Dynatrace/dynatrace-operator
# Use new GORM client and DB tables in CSI Driver

## Description

Use the new GORM client interface _everywhere_ on the CSI driver, replacing the old SQLite interface.

Jira ticket -> https://dt-rnd.atlassian.net/browse/K8S-9328

## How can this be tested?

- Unit tests
- E2E tests triggered on branch ~~([link here](https://github.com/Dynatrace/dynatrace-operator/actions/runs/8687104816))~~ ([latest execution of E2E](https://github.com/Dynatrace/dynatrace-operator/actions/runs/8831720306/job/24247509841))
- Deploy dynakube with branch changes and check for any error
- - Try to upgrade the codemodules version so a new entry is stored in CSI driver DB

**How to check csi driver DB:**
```sh
// replace with the name of your csi-driver pod
kubectl cp dynatrace-oneagent-csi-driver-4nmkm:/data/csi.db csi.sqlite
```
```sh
sqlite3 csi.sqlite
// takes you to sqlite prompt
// check existing tables
.tables
// query for values
select * from volume_meta;
```

![image](https://github.com/Dynatrace/dynatrace-operator/assets/61692514/9b74951d-a9ad-45dc-a30c-0e9497a0a25f)

## Checklist
